### PR TITLE
Skip evaluating anonymous inner classes when merging modules and interfaces

### PR DESCRIPTION
**Pull request by JoelWilcox**

### CODE REVIEW

File: compiler/src/test/java/com/squareup/anvil/compiler/InterfaceMergerTest.kt [hunk 1]
```diff
@@ -622,4 +623,26 @@ class InterfaceMergerTest(
       assertThat(componentInterface extends secondContributingInterface).isTrue()
     }
   }
+
+  @Test fun `anonymous inner classes should not be evaluated`() {
```
Comment (JoelWilcox) [inline compiler/src/test/java/com/squareup/anvil/compiler/InterfaceMergerTest.kt:627]: This case came up initially from an `InterfaceMerger`-based stacktrace, although technically this test isn't specific to `InterfaceMerger`. Open to giving it a new home if you have a better suggestion @vRallev

File: compiler/src/main/java/com/squareup/anvil/compiler/InterfaceMerger.kt [hunk 2]
```diff
@@ -30,6 +31,10 @@ internal class InterfaceMerger(
     thisDescriptor: ClassDescriptor,
     supertypes: MutableList<KotlinType>
   ) {
+    // If we're evaluating an anonymous inner class, it cannot contribute anything and will cause
```
Comment (vRallev) [inline compiler/src/main/java/com/squareup/anvil/compiler/InterfaceMerger.kt:34]: > it cannot contribute anything.

I think this should be "it cannot merge anything".

File: compiler/src/test/java/com/squareup/anvil/compiler/InterfaceMergerTest.kt (hunk 1 above)
Comment (vRallev) [reply compiler/src/test/java/com/squareup/anvil/compiler/InterfaceMergerTest.kt:627]: I think that's fine.

File: compiler/src/test/java/com/squareup/anvil/compiler/InterfaceMergerTest.kt [hunk 3]
```diff
@@ -622,4 +623,26 @@ class InterfaceMergerTest(
...
+    compile(
+      """
+      package com.squareup.test
+
+      class NonContributingClass {
+        private val anonymousDoSomething = object : DoSomethingInterface {
+          override fun doSomething() { 
+            // unimportant impl
+          }
```
Comment (vRallev) [inline compiler/src/test/java/com/squareup/anvil/compiler/InterfaceMergerTest.kt:636]:
```suggestion
          override fun doSomething() = Unit
```
We use that to indicate that the body is empty on purpose.

**Review by vRallev — APPROVED** [review]

**Pull request merged**